packages/runtime/src/utils.ts

```diff
@@ -21,29 +24,39 @@ export function parseDependencies(dependencies: any): Dependency[] {
   });
 }
 
+const mapDependencyToValue = (dependency: Dependency): string => {
+  if (typeof dependency === 'number') {
+    return dependency.toString();
+  }
+  if (typeof dependency === 'string') {
+    return dependency;
+  }
+  return dependency.uniqueId;
+}
+
 export const calculateCacheKey = (modulePath: string, dependencies: Dependency[]): string => {
-  const dependenciesString = dependencies.map((dependency) => {
-    if (typeof dependency === 'number') {
-      return dependency.toString();
-    }
-    if (typeof dependency === 'string') {
-      return dependency;
-    }
-    return dependency.uniqueId;
-  }).join(',');
+  const dependenciesString = dependencies.map(mapDependencyToValue).join(',');
 
   return `${modulePath}::${dependenciesString}`;
 }
 
-export type ServerComputationFunction = (...dependencies: any[]) => any;
-export type ClientComputationFunction = () => Promise<{ default: (...dependencies: any[]) => any }>
+export type SSRComputationModule<TResult> = {
+  compute: (...dependencies: Dependency[]) => TResult | NoResultType;
```

Comment:
The `compute` function can return `NoResult` to tell that the result is not loaded yet. So, the plugin will keep the previously cached result until the new result is fetched and passed to the plugin throw the `next` function